tools/system_libs.py

```diff
@@ -744,6 +779,9 @@ def do_create():
     for d in lib.deps:
       add_library(system_libs_map[d])
 
+    for d in lib.js_deps:
+      shared.Settings.EXPORTED_FUNCTIONS.append('_%s' % d)
```

Comment:
Can you explain this a litte? It looks like it just used for memalign?  Is memalign implemented in JS?   Why to do need js_deps now but we've managed to get away without them previously?

Reply:
This is `__syscall192` (a JS function) depending on `memalign` (a C function), but we only add back dependencies to user code, not libraries.

Here, the sanitizer common library depends on `__syscall192`, but `memalign` is not pulled in because the user code doesn't use `__syscall192`.

I am not sure if this is the best approach, but I am doing this for now so that it at least works.